tools/chapel-py/src/python-type-helper.h

```diff
@@ -0,0 +1,45 @@
+/*
+ * Copyright 2023-2025 Hewlett Packard Enterprise Development LP
+ * Other additional copyright holders may be indicated within.
+ *
+ * The entirety of this work is licensed under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ *
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#ifndef CHAPEL_PY_PYTHON_TYPE_HELPER_H
+#define CHAPEL_PY_PYTHON_TYPE_HELPER_H
+
+#include "PythonWrapper.h"
+
+static void call_tp_free(PyTypeObject* type, PyObject* self) {
+  auto tp_free = (void (*)(PyObject*))(PyType_GetSlot(type, Py_tp_free));
+  if (!tp_free) {
+    PyErr_SetString(PyExc_RuntimeError, "Could not free object");
+  } else {
+    tp_free(self);
+  }
+}
+
+static const char* getTypeName(PyTypeObject* obj) {
+  PyObject* name_attr = PyObject_GetAttrString((PyObject*) obj, "__name__");
+  if (!name_attr) {
+    PyErr_SetString(PyExc_RuntimeError, "Could not get type name");
+    return nullptr;
+  }
+  const char* name = PyUnicode_AsUTF8AndSize(name_attr, NULL);
```

Comment:
This is new, and previously not required by `chapel-py`. _our_ types are definitely not unicode-named; can we use non-unicode functions here, thus saving us from using 3.10+?

Reply:
We do use `PyUnicode_AsUTF8AndSize` elsewhere, so removing it here wouldn't help that requirement. 

We had to stop using `->tp_name`, as PyTypeObject is now opaque. Another option here is `PyType_GetName`, but that requires 3.11+ and would still need `PyUnicode_AsUTF8AndSize`.

In rereading the docs, I realized that we may be able to use `PyArg_Parse` as a replacement for `PyUnicode_AsUTF8AndSize`? I will try it

Reply:
We can use PyArg_Parse :grin:. I think that opens up earlier version of Python for the stable API